sdk/src/lock.c

```diff
@@ -240,7 +240,10 @@ int fhs_lock(const char *filename, int pid) {
 
   sprintf(lockinfo, "%10d\n", (int) getpid());
   printf("fhs_lock: creating lockfile: %s\n", lockinfo);
-  write(fd, lockinfo, 11);
+  if (write(fd, lockinfo, 11) < 0) {
```

Comment:
The number of bytes written may be less than count in some cases. Can we also check it along with failure check ?